cfgov/unprocessed/js/molecules/Multiselect.js

```diff
@@ -0,0 +1,429 @@
+'use strict';
+
+// Required polyfills for <IE9.
+require( '../modules/polyfill/query-selector' );
+require( '../modules/polyfill/event-listener' );
+require( '../modules/polyfill/class-list' );
+
+// Required modules.
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+var arrayHelpers = require( '../modules/util/arrays' );
+var typeCheckers = require( '../modules/util/type-checkers' );
+
+
+/**
+ * Multiselect
+ * @class
+ *
+ * @classdesc Initializes a new Multiselect molecule.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the molecule.
+ * @returns {Object} A Multiselect instance.
+ */
+function Multiselect( element ) { // eslint-disable-line max-statements, inline-comments, max-len
+
+  // Bitwise flags for the state of this Expandable.
+  var COLLAPSED = 0;
+  var COLLAPSING = 1;
+  var EXPANDING = 2;
+  var EXPANDED = 3;
+
+  // Constants for direction
+  var DIR_PREV = 'prev';
+  var DIR_NEXT = 'next';
+
+  // Constants for key binding.
+  var KEY_RETURN = 13;
+  var KEY_ESCAPE = 27;
+  var KEY_UP = 38;
+  var KEY_DOWN = 40;
+  var KEY_TAB = 9;
+
+  // Internal vars.
+  var _dom = element;
+  var _state = COLLAPSED;
+  var _minChars = 3;
+  var _maxItems = 10;
+  var _index = -1;
+  var _isBlurSkipped = false;
+  var _optionData = [];
+  var _filtered = [];
+  var _selections = [];
+  var _name;
+  var _options;
+  var _placeholder;
+
+  //Markup elems.
+  var _container;
+  var _choices;
+  var _header;
+  var _search;
+  var _fieldset;
+  var _list;
+
+  function init() {
+    _name = _dom.name;
+    _options = _dom.options || [];
+    _placeholder = _dom.getAttribute( 'placeholder' );
+
+    _sanitizeList( _options );
+
+    if ( _optionData.length > 0 ) {
+      _populateMarkup();
+      _bindEvents();
+    }
+
+    return this;
+  }
+
+  function expand() {
+    _container.classList.add( 'active' );
+    _fieldset.setAttribute( 'visibility', 'visible' );
+  };
```

Comment:
You could add `return this` to the end of `expand()` and `collapse()` so that `Multiselect` supports method chaining. In use this would look like:

```
var Multiselect  = require( '/path/to/module/Multiselect' );
var multiselect = new Multiselect( document.querySelector( 'm-multiselect' ) ).init().collapse().expand();
```

Perhaps it doesn't make sense to chain methods here since collapsing and expanding the menu sequentially doesn't make sense. In that case you could remove the `return this` from the `init()` method too for consistent, which would make the implementation look like:

```
var Multiselect  = require( '/path/to/module/Multiselect' );
var multiselect = new Multiselect( document.querySelector( 'm-multiselect' ) );
multiselect.init();
```

CodeClimate probably prefers that way.